hoomd/mpcd/pytest/test_collide.py

```diff
@@ -130,3 +131,142 @@ def test_run(self, small_snap, simulation_factory, cls, init_args):
+    )
+    def test_rigid_collide_linear(
+        self,
+        one_particle_snapshot_factory,
+        simulation_factory,
+        cls,
+        init_args,
+        rigid_velo,
+        rigid_angmom,
+        rigid_pos,
+        rigid_def,
+        rigid_properties,
+    ):
+        if "kT" not in init_args:
+            init_args["kT"] = 1.0
+        mpcd_N = len(rigid_def["constituent_types"])
+        rng = np.random.default_rng(seed=42)
+        mpcd_velo = rng.normal(0.0, np.sqrt(init_args["kT"]), (mpcd_N, 3))
+        mpcd_velo -= np.mean(mpcd_velo, axis=0)
+
+        # create simulation
+        initial_snap = one_particle_snapshot_factory(
+            particle_types=["A", "B"], position=rigid_pos
+        )
+        if initial_snap.communicator.rank == 0:
+            initial_snap.particles.moment_inertia[:] = [rigid_properties["inertia"]]
+            initial_snap.particles.mass[:] = [rigid_properties["mass"][0]]
+            initial_snap.particles.velocity[:] = [rigid_velo]
+            initial_snap.particles.angmom[:] = [rigid_angmom]
+
+        sim = simulation_factory(initial_snap)
+        sim.seed = 5
+
+        rigid = hoomd.md.constrain.Rigid()
+        rigid.body["A"] = rigid_def
+        rigid.create_bodies(sim.state)
+
+        intermed_snap = sim.state.get_snapshot()
+        if intermed_snap.communicator.rank == 0:
+            # add mass of constituents
+            flags = (
+                intermed_snap.particles.typeid
+                == intermed_snap.particles.types.index("B")
+            )
+            intermed_snap.particles.mass[flags] = rigid_properties["mass"][flags]
+            intermed_snap.wrap()
+
+            # place the mpcd particles on top of constituents
+            intermed_snap.mpcd.N = mpcd_N
+            intermed_snap.mpcd.types = ["C"]
+            intermed_snap.mpcd.position[:] = intermed_snap.particles.position[flags]
+            intermed_snap.mpcd.velocity[:] = mpcd_velo
+        sim.state.set_snapshot(intermed_snap)
+
+        sim.operations.integrator = hoomd.mpcd.Integrator(
+            dt=0, integrate_rotational_dof=True, rigid=rigid
+        )
+
+        sim.operations.integrator.collision_method = cls(
+            period=1,
+            embedded_particles=hoomd.filter.Rigid(flags=("constituent",)),
+            **init_args,
+        )
+
+        # run simulation
+        sim.run(1)
+        new_snap = sim.state.get_snapshot()
+        if new_snap.communicator.rank == 0:
+            assert np.array_equal(rigid_properties["mass"], new_snap.particles.mass)
+            new_velo = new_snap.particles.velocity
+            # the central particle speed should change despite not being in the filter
+            assert not np.any(np.isclose(new_velo[0], rigid_velo))
+
+            # constituent velocities average to central velocity
+            calculated_central_speed = np.mean(new_velo, axis=0)
+            assert np.allclose(new_velo[0], calculated_central_speed)
```

Comment:
Shouldn't they all be exactly the same since it is a rigid body?

Reply:
When I checked their actual values, the constituents did have different velocities due to the additional angular momentum.

Reply:
See my comment above about needing to update the constituent particle velocities after the transfer happens. This test is passing but for the wrong reason.